### PR TITLE
Version 2 enhancements

### PR DESCRIPTION
**Pull request by theurichde**

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out the repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Push to Docker Hub
         uses: docker/build-push-action@v1
```
Comment (stat1981) [inline .github/workflows/docker-image.yml:16]: Can be updated to v3

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
 
-# Lean container
-FROM alpine:latest
+FROM alpine:3.16
 RUN apk --no-cache add ca-certificates
 WORKDIR /app
-COPY --from=0 /build/cmd/aws-signing-proxy/aws-signing-proxy .
+COPY --from=builder /build/cmd/aws-signing-proxy/aws-signing-proxy .
 
 RUN addgroup -S proxy && adduser -S proxy -G proxy
```
Comment (stat1981) [inline Dockerfile:11]: Dont make to much RUNs, DOCKERFILE best pratice is too use one with &&\

File: README.md [hunk 3]
```diff
@@ -13,46 +13,43 @@ Supported AWS credentials:
   an [AWS secrets engine & sts-assumerole](https://www.vaultproject.io/docs/secrets/aws#sts-assumerole)
 * Fetching short-lived credentials from AWS via a OAuth2 authorization server
   and [OpenID Connect (OIDC)](https://openid.net/connect/)
+  * Additionally, you can fetch these credentials asynchronously
```
Comment (stat1981) [inline README.md:16]: Here is the how can this feature be activated missing.

File: README.md [hunk 4]
```diff
@@ -13,46 +13,43 @@ Supported AWS credentials:
...
-In addition to the proxy you may also use `vault-env-cred-provider` as an
-[credential provider for AWS tooling](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html)
-.
+### Breaking Changes
 
-❗NOTE: the provided pre-built macOS binaries might fail with name resolution issues on your apple machine if you are
-using a (corporate) VPN. This will not occur on linux/windows/docker. If you are affected: either use the provided
-docker image or build the binaries on your machine.
+* Command line flags are now kebab-case to be POSIX style guide compliant
+* `Health Port` is now called `Mgmt Port`
```
Comment (stat1981) [inline README.md:32]: Here is a part in the README missing, how to configure the MGMT Port and the Prometheus endpoint.

**Review by stat1981 — APPROVED** [review]

**Review by stat1981 — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (theurichde) [reply README.md:32]: Good point

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (theurichde) [reply .github/workflows/docker-image.yml:16]: It's on the agenda. The upgrade introduces some breaking changes I don't want to  deal with right now

**Review by roechi — DISMISSED** [review]

**Review by stat1981 — APPROVED** [review]

**Pull request merged**